src/change_stream.ts

```diff
@@ -648,24 +593,18 @@ function processError(changeStream: ChangeStream, error?: AnyError, callback?: C
 
   // otherwise, raise an error and close the change stream
   function unresumableError(err: AnyError) {
-    if (!callback) {
-      changeStream.emit(ChangeStream.ERROR, err);
-      changeStream.emit(ChangeStream.CLOSE);
-    }
-    processResumeQueue(changeStream, err);
-    changeStream.closed = true;
+    if (!callback) changeStream.emit(ChangeStream.ERROR, err);
+    changeStream.close(() => processResumeQueue(changeStream, err));
   }
 
   if (cursor && isResumableError(error as MongoError, maxWireVersion(cursor.server))) {
     changeStream.cursor = undefined;
 
     // stop listening to all events from old cursor
-    [
-      ChangeStreamCursor.DATA,
-      ChangeStreamCursor.CLOSE,
-      ChangeStreamCursor.END,
-      ChangeStreamCursor.ERROR
-    ].forEach(event => cursor.removeAllListeners(event));
+    [CursorStream.DATA, CursorStream.CLOSE, CursorStream.END, CursorStream.ERROR].forEach(event =>
```

Comment:
similar comment to above, no need to iterate over all of these if the cursor stream doesn't exist.